src/Raven.Server/Rachis/ClusterMaintenanceSlave.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Replication.Messages;
+using Raven.Server.Documents.TcpHandlers;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Rachis
+{
+    public class ClusterMaintenanceSlave : IDisposable
+    {
+        private readonly TcpConnectionOptions _tcp;
+        private readonly ServerStore _server;
+        private CancellationToken _token;
+        private readonly CancellationTokenSource _cts;
+        private Task _collectingTask;
+        private Dictionary<string,DatabaseStatusReport> _cachedReport = new Dictionary<string, DatabaseStatusReport>();
```

Comment:
* Why do we want to keep that information? That db is down, so it obviously not active.
* We can figure out at least some of the information from the cached db stats we already have (see the `/stats` endpoint for the studio.
* Either case, it shouldn't be in a field on a class that might be exposed to multi threading.